.gitlab-ci.yml

```diff
@@ -327,3 +327,110 @@ publish:docker-image:automatic:
   rules:
     - if: '$CI_COMMIT_BRANCH =~ /^(master|staging|production|feature-.+|[0-9]+\.[0-9]+\.([0-9]+|x))$/'
   extends: .template:publish:docker-image
+
+####################
+# Hardware testing #
+####################
+
+test:hardware:build:
+  <<: *convert_raspbian
+  variables:
+    RASPBERRYPI_PLATFORM: raspberrypi4
+  when: manual
+  stage: build
+  script:
+    # Create the custom needed configurations
+    - |
+      cat <<-EOF > configs/hardware-testing-conf
+      MENDER_COMPRESS_DISK_IMAGE="none"
+      MENDER_ARTIFACT_COMPRESSION="none"
+      # Install the mender-client
+      MENDER_CLIENT_INSTALL="y"
+      # Enable sshd
+      function enable_sshd () {
+              log_warn "Enabling sshd by default"
+              touch work/boot/ssh
+      }
+      PLATFORM_MODIFY_HOOKS+=(enable_sshd)
+      # Allow root ssh logins
+      function allow_root_ssh_logins() {
+              log_warn "Allowing ssh root logins"
+              echo "PermitRootLogin yes" >> /etc/ssh/sshd_config
+      }
+      PLATFORM_MODIFY_HOOKS+=(allow_root_ssh_logins)
+      EOF
+    # Configure the ssh key permissions
+    - source scripts/test/test-utils.sh
+    - prepare_ssh_keys
+    # Convert the image for the hardware tester
+    - env MENDER_ARTIFACT_NAME=${RASPBIAN_NAME}-mender
+      ./docker-mender-convert --disk-image ${RASPBIAN_NAME}.img
+      --config configs/${RASPBERRYPI_PLATFORM}_config
+      --config configs/images/raspberrypi_raspbian_config
+      --config configs/hardware-testing-conf
+      --overlay tests/ssh-public-key-overlay
+    # Collect artifacts.
+    - mv deploy ${RASPBERRYPI_PLATFORM}
+    - tar czf ${RASPBERRYPI_PLATFORM}.tar.gz ${RASPBERRYPI_PLATFORM}
+    # Upload to temporary S3 bucket
+    - mender_ci_save_tmp_artifact ${RASPBERRYPI_PLATFORM}.tar.gz
+  artifacts:
+    paths:
+      - checksums
+
+test:hardware:flash:
+  needs:
+    - job: test:hardware:build
+      artifacts: true
+  variables:
+    RASPBERRYPI_PLATFORM: raspberrypi4
+  tags:
+    - hw
+  stage: build
+  script:
+    - eval "$(curl https://raw.githubusercontent.com/mendersoftware/mendertesting/master/mender-ci-common.sh)"
+    # Fetch artifacts from temporary S3 bucket
+    - mender_ci_load_tmp_artifact ${RASPBERRYPI_PLATFORM}.tar.gz
+    # Flash and boot the device under test
+    - slave-control mode flash
+    - tar -xvzf ${RASPBERRYPI_PLATFORM}.tar.gz --wildcards --no-anchored '*.img' --to-stdout | dd of=/dev/sdmux bs=8M
+    - slave-control mode test
+
+test:hardware:acceptance:
+  <<: *test_acceptance
+  variables:
+    RASPBERRYPI_PLATFORM: raspberrypi4
+    SSH_JUMP_HOST_IP: 146.185.128.173
+    SSH_JUMP_HOST_PORT: "1234"
```

Comment:
Should these two be configured as variables on Gitlab instead?

Reply:
Hmm, maybe. I liked the visibility, but GitLab would be more flexible...

Alright, I'll add it to GitLab